examples/src/main/java/org/apache/ignite/examples/datagrid/store/CacheLoadOnlyStoreExample.java

```diff
@@ -0,0 +1,192 @@
+import org.apache.ignite.internal.util.typedef.T2;
+import org.apache.ignite.lang.IgniteBiTuple;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Example of how to load data from CSV file.
+ *
+ * Note that it aims to illustrate the use of {@link CacheLoadOnlyStoreAdapter}
+ * and does not contains thorough error checking and validation.
+ */
+public class CacheLoadOnlyStoreExample {
+    /** Cache name. */
+    private static final String CACHE_NAME = CacheLoadOnlyStoreExample.class.getSimpleName();
+
+    public static void main(String[] args) throws Exception {
+        try (Ignite ignite = Ignition.start("examples/config/example-ignite.xml")) {
+            System.out.println();
+            System.out.println(">>> CacheLoadOnlyStoreExample started.");
+
+            ProductLoader productLoader = new ProductLoader("product.csv");
+
+            productLoader.setThreadsCount(2);
+            productLoader.setBatchSize(10);
+            productLoader.setBatchQueueSize(1);
+
+            try (IgniteCache<String, Product> cache = ignite.getOrCreateCache(cacheConfiguration(productLoader))) {
+                // load data.
+                cache.loadCache(null);
+
+                System.out.println(">>> Loaded number of items: " + cache.size(CachePeekMode.PRIMARY));
+
+                System.out.println(">>> Product id1's data: " + cache.get("id1"));
+            }
+            finally {
+                // Distributed cache could be removed from cluster only by #destroyCache() call.
+                ignite.destroyCache(CACHE_NAME);
+            }
+        }
+    }
+
+    /**
+     * Creates cache configurations for the loader.
+     *
+     * @return {@link CacheConfiguration}.
+     */
+    private static CacheConfiguration cacheConfiguration(ProductLoader productLoader) {
+        CacheConfiguration cacheCfg = new CacheConfiguration();
+
+        cacheCfg.setCacheMode(CacheMode.PARTITIONED);
+        cacheCfg.setName(CACHE_NAME);
+
+        cacheCfg.setReadThrough(true);
+        cacheCfg.setWriteThrough(true);
+        cacheCfg.setLoadPreviousValue(true);
+
+        // provide the loader.
+        cacheCfg.setCacheStoreFactory(new FactoryBuilder.SingletonFactory(productLoader));
+
+        return cacheCfg;
+    }
+
+    /**
+     * Csv data loader for product data.
+     */
+    private static class ProductLoader extends CacheLoadOnlyStoreAdapter<String, Product, String> implements Serializable {
+        /** Csv file name. */
+        final String csvFileName;
+
+        /** Constructor. */
+        ProductLoader(String csvFileName) {
+            this.csvFileName = csvFileName;
+        }
+
+        /** {@inheritDoc} */
+        @Override protected Iterator<String> inputIterator(@Nullable Object... args) throws CacheLoaderException {
+            Scanner scanner;
+
+            try {
+                scanner = new Scanner(new File(this.getClass().getClassLoader().getResource(csvFileName).getFile()));
```

Comment:
I got NPE at this line. Please modify this snippet of the code to the following and check the example against 'master' branch where I got the exeption

                URL path = getClass().getClassLoader().getResource(csvFileName);

                if (path == null)
                    throw new CacheLoaderException("Failed to open the source file: " + csvFileName);

                scanner = new Scanner(new File(path.getFile()));